app/src/main/java/com/andrew/tasky/auth/presentation/screens/register/RegisterFragment.kt

```diff
@@ -2,45 +2,87 @@ package com.andrew.tasky.auth.presentation.screens.register
 
 import android.os.Bundle
 import android.view.View
+import android.widget.Toast
 import androidx.core.view.isVisible
 import androidx.core.widget.addTextChangedListener
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.ViewModelProvider
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
 import androidx.navigation.NavController
 import androidx.navigation.Navigation
 import com.andrew.tasky.R
 import com.andrew.tasky.agenda.util.collectLatestLifecycleFlow
+import com.andrew.tasky.auth.AuthResult
+import com.andrew.tasky.auth.presentation.screens.login.LoginFragmentDirections
 import com.andrew.tasky.auth.util.NameValidator
 import com.andrew.tasky.databinding.FragmentRegisterBinding
 
 class RegisterFragment : Fragment(R.layout.fragment_register) {
 
-    private lateinit var viewModel: RegisterViewModel
+    private val viewModel: RegisterViewModel by viewModels()
     private lateinit var binding: FragmentRegisterBinding
     private lateinit var navController: NavController
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding = FragmentRegisterBinding.bind(view)
         navController = Navigation.findNavController(view)
-        viewModel = ViewModelProvider(this).get(RegisterViewModel::class.java)
 
-        binding.backButton.setOnClickListener() {
-            navController.popBackStack()
-        }
+        binding.apply {
 
-        binding.nameEditText.addTextChangedListener {
-            if (it != null) {
-                viewModel.setIsNameValid(NameValidator().validate(it.toString()))
+            backButton.setOnClickListener() {
+                navController.popBackStack()
+            }
+
+            nameEditText.addTextChangedListener {
+                if (it != null) {
+                    viewModel.setIsNameValid(NameValidator().validate(it.toString()))
+                }
             }
-        }
 
-        binding.getStartedButton.setOnClickListener {
-            binding.passwordTextField.isPasswordValid()
+            getStartedButton.setOnClickListener {
+                if (
+                    passwordTextField.isPasswordValid() &&
+                    viewModel.isNameValid.value &&
+                    emailTextField.isValid()
```

Comment:
Would it be possible to move this condition in the ViewModel in `register()`?
I like to keep the UI as dumb as possible about such logic, so it only needs to display UI and send actions to the VM. That's helpful because you can easier test logic of a ViewModel than logic in the UI (here you would need to run an actual UI test that simulates clicks etc)